chef/cookbooks/provisioner/libraries/repositories.rb

```diff
@@ -17,69 +17,20 @@
 class Provisioner
   class Repositories
     class << self
-      def inspect_repos(node)
-        unless node.roles.include? "provisioner-server"
-          raise "Internal error: inspect_repos method should only be called on provisioner-server node."
-        end
-
-        common_available, cloud_available, hae_available, storage_available = nil
-        repos_from_databag = Chef::DataBag.load("repositories").keys rescue []
-
-        case node[:platform]
-        when "suse"
-          repos_from_databag.each do |repo_from_databag|
-            dbi = Chef::DataBagItem.load("repositories", repo_from_databag)
-            case dbi["product_name"]
-            when "common"
-              common_available = true
-            when "cloud"
-              cloud_available = true
-            when "hae"
-              hae_available = true
-            when "storage"
-              storage_available = true
-            end
-          end
-
-          # set an attribute about available repos so that cookbooks and crowbar
-          # know that HA can be used
-          # know that SUSE_Storage can be used
-          # know that OpenStack can be used
-          node_set = false
-          node.set[:provisioner][:suse] ||= {}
-          if node[:provisioner][:suse][:common_available] != common_available
-            node.set[:provisioner][:suse][:common_available] = common_available
-            node_set = true
-          end
-          if node[:provisioner][:suse][:cloud_available] != cloud_available
-            node.set[:provisioner][:suse][:cloud_available] = cloud_available
-            node_set = true
-          end
-          if node[:provisioner][:suse][:hae_available] != hae_available
-            node.set[:provisioner][:suse][:hae_available] = hae_available
-            node_set = true
-          end
-          if node[:provisioner][:suse][:storage_available] != storage_available
-            node.set[:provisioner][:suse][:storage_available] = storage_available
-            node_set = true
-          end
-          if node_set
-            node.save
-          end
-        end
-      end
-
       # This returns a hash containing the data about the repos that must be
       # used on nodes; optional repos (such as HA) will only be returned if
       # they can be used.
       def get_repos(platform, version)
         repositories = {}
-        repos_from_databag = Chef::DataBag.load("repositories").keys rescue []
-        repos_from_databag.each do |reponame|
-          dbi = Chef::DataBagItem.load("repositories", reponame)
-          if dbi["platform"] == "#{platform}-#{version}"
-            repositories[dbi["name"]] = { url: dbi["url"] }
-          end
+
+        begin
+          repos_db = Chef::DataBag.load("crowbar/repositories")
+        rescue Net::HTTPServerException
+          repos_db = {}
+        end
+
+        repos_db.fetch("#{platform}-#{version}", {}).each do |id, config|
```

Comment:
but as we already found out, `Chef::DataBag.load` doesn't return nil in any case, that is why i was wondering